### PR TITLE
Allow updating dimmer brightness when lamp is off

### PR DESCRIPTION
**Pull request by abstractionnl**
This change adds a number input that can be used to set the brightness of the dimmer when the light is off.
Home Assistant by default only supports setting the dimmer brightness when turning the light on. This makes it not possible to set the brightness when using the physical switch attached to the dimmer before the light is turned on. For example when setting the brightness to low at night using an automation.

### CODE REVIEW

**Comment by bieniu** [comment]
Does the device's web interface allow you to adjust the brightness when the light is off?

**Comment by abstractionnl** [comment]
afbeelding

See above

You can call GET http://192.168.3.38/light/0?brightness=74 to update the brightness via the API, or send {" brightness" : 74} to the light/0/set MQTT topic.

**Comment by bieniu** [comment]
Shelly team says this is an intended possibility. Please fix the lint errors so we can continue.

File: python_scripts/shellies_discovery.py [hunk 1]
```diff
@@ -3393,6 +3410,46 @@ def mqtt_publish(topic, payload, retain, json=False):
 
         mqtt_publish(config_topic, payload, retain)
 
+    # white light numbers
```
Comment (bieniu) [inline python_scripts/shellies_discovery.py:3413]:
```suggestion
    # light numbers
```

File: python_scripts/shellies_discovery.py [hunk 2]
```diff
@@ -3393,6 +3410,46 @@ def mqtt_publish(topic, payload, retain, json=False):
...
+    for number, number_options in light_numbers.items():
+        config_topic = (
+            f"{disc_prefix}/number/{dev_id}-white-{number}-{light_id}/config".encode(
```
Comment (bieniu) [inline python_scripts/shellies_discovery.py:3416]:
```suggestion
            f"{disc_prefix}/number/{dev_id}-{light_id}-{number}/config".encode(
```

File: python_scripts/shellies_discovery.py [hunk 3]
```diff
@@ -3393,6 +3410,46 @@ def mqtt_publish(topic, payload, retain, json=False):
...
+            ),
+            KEY_COMMAND_TEMPLATE: number_options[KEY_COMMAND_TEMPLATE].format(),
+            KEY_MAX: number_options[KEY_MAX],
+            KEY_MIN: number_options[KEY_MIN],
+            KEY_STEP: number_options[KEY_STEP],
+            KEY_STATE_TOPIC: number_options[KEY_STATE_TOPIC].format(light_id=light_id),
+            KEY_VALUE_TEMPLATE: number_options[KEY_VALUE_TEMPLATE],
+            KEY_UNIT: number_options[KEY_UNIT],
+            KEY_ENABLED_BY_DEFAULT: str(number_options[KEY_ENABLED_BY_DEFAULT]).lower(),
+            KEY_UNIQUE_ID: f"{dev_id}-white-{number}-{light_id}".lower(),
```
Comment (bieniu) [inline python_scripts/shellies_discovery.py:3434]:
```suggestion
            KEY_UNIQUE_ID: f"{dev_id}-{light_id}-{number}".lower(),
```

File: python_scripts/shellies_discovery.py [hunk 4]
```diff
@@ -3393,6 +3410,46 @@ def mqtt_publish(topic, payload, retain, json=False):
...
+    # white light numbers
+    for number, number_options in light_numbers.items():
+        config_topic = (
+            f"{disc_prefix}/number/{dev_id}-white-{number}-{light_id}/config".encode(
+                "ascii", "ignore"
+            ).decode("utf-8")
+        )
+
+        payload = {
+            KEY_NAME: f"{device_name} {format_entity_name(number)} {light_id}",
```
Comment (bieniu) [inline python_scripts/shellies_discovery.py:3422]:
```suggestion
            KEY_NAME: f"{device_name} Light {light_id} {number}",
```

**Pull request merged**